v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -166,29 +135,91 @@ export async function main(cliArguments: string[]) {
 
     console.log("Error running the task:", error.message);
 
-    if (showStackTraces) {
+    if (hhGlobalParams.showStackTraces) {
       console.log("");
       console.error(error);
     }
   }
 }
 
-function parseGlobalArguments(
-  _globalParamsIndex: GlobalParameterMap,
-  _cliArguments: string[],
-  _usedCliArguments: boolean[],
-): Partial<GlobalArguments> {
-  // TODO: Parse the global arguments
-  // - Parse the global params, skipping the processed entries
-  // - At each stage validate the value according to its type
-  // - If a global param is boolean and its default is false, its value is not
-  //   necessary. That means that if the immediatly next value is `"true"` or
-  //   `"false"`, we should use it, but if it's not we just don't, and consider
-  //   its value `true`.
-  // - Mark all the used entries in usedCliArguments
-  // - Do not resolve defaults here, the HRE does that.
-  // - Return the values
-  return {};
+export async function parseInitialHardhatParameters(
+  cliArguments: string[],
+  usedCliArguments: boolean[],
+) {
+  let configPath: string | undefined;
+  let showStackTraces: boolean = false; // true if ci
+  let help: boolean = false;
+  let version: boolean = false;
+
+  for (let i = 0; i < cliArguments.length; i++) {
+    const arg = cliArguments[i];
+
+    if (arg === "--config") {
+      usedCliArguments[i] = true;
+
+      if (configPath !== undefined) {
+        throw new HardhatError(HardhatError.ERRORS.ARGUMENTS.DUPLICATED_NAME, {
+          name: "--config",
+        });
+      }
+
+      if (
+        usedCliArguments[i + 1] === undefined ||
```

Comment:
can we simplify this expression to `usedCliArguments[i + 1] !== false`?

Reply:
Yes, it would work, but in a previous PR Pato mentioned that we should be explicit when checking for out of bounds checks

Reply:
ohh got it. Then you can ignore it!

Reply:
This would be a bit less confusing if it were `i < usedCliArguments.length &&`

Reply:
I'd prefer that. The current check doesn't make the intention explicit.